lib/helper.c

```diff
@@ -44,8 +44,16 @@ long int unshield_get_path_max(Unshield* unshield)
 char *unshield_get_base_directory_name(Unshield *unshield) {
     long int path_max = unshield_get_path_max(unshield);
     char *p = strrchr(unshield->filename_pattern, '/');
+#ifdef WIN32
+    char *pbs = strrchr(unshield->filename_pattern, '\\');
+#endif
     char *dirname = malloc(path_max);
 
+#ifdef WIN32
+    if (NULL != pbs && (NULL == p || pbs > p))
+      p = pbs;
+#endif
+
```

Comment:
Could this "find last slash or backslash on windows" logic be extracted to a helper function and reused in both places?